packages/splits-sdk/src/utils/index.ts

```diff
@@ -115,3 +132,99 @@ export const getRecoupTranchesAndSizes = async (
 
   return [recoupTranches, sizes]
 }
+
+export const getAddressAndAllocationFromRecipients = (
+  recipients: SplitRecipient[],
+): { recipientAddresses: Address[]; recipientAllocations: bigint[] } => {
+  return {
+    recipientAddresses: recipients.map(
+      (recipient) => recipient.address,
+    ) as Address[],
+    recipientAllocations: recipients.map((recipient) =>
+      getBigIntFromPercent(recipient.percentAllocation),
+    ),
+  }
+}
+
+export const MAX_DISTRIBUTION_INCENTIVE = 6.5535
+
+export const getValidatedSplitV2Config = (
+  recipients: SplitRecipient[],
+  distributorFeePercent: number,
+  totalAllocationPercent?: number,
+): {
+  recipientAddresses: Address[]
+  recipientAllocations: bigint[]
+  distributionIncentive: number
+  totalAllocation: bigint
+} => {
+  const { recipientAddresses, recipientAllocations } =
+    getAddressAndAllocationFromRecipients(recipients)
+
+  if (distributorFeePercent > MAX_DISTRIBUTION_INCENTIVE)
+    throw new InvalidDistributorFeePercentErrorV2(distributorFeePercent)
+  const distributionIncentive = getNumberFromPercent(distributorFeePercent)
+
+  const calculatedTotalAllocation = recipientAllocations.reduce((a, b) => a + b)
```

Comment:
just be careful of javascript floating point precision. i.e. might want to round this to the nearest .00001 (or whatever is one smaller than the smallest decimal we accept).

otherwise you might end up with some b.s. like 99.99999999998 instead of 100